import "stream.proto";
import "cert.proto";
import "signature.proto";

message Claim {
    enum Version {
        UNKNOWN_VERSION = 0;
        _0_0_1 = 1;
    }
    required Version version = 1;

    enum ClaimType {
        UNKNOWN_CLAIM_TYPE = 0;
        streamClaim = 1;
        certificateClaim = 2;
    }
    required ClaimType claimType = 2;
    optional Stream stream = 3;
    optional Cert certificate = 4;
    optional Signature publisherSignature = 5;
}
